features/64-chat-offline-cache/api.md

```diff
@@ -0,0 +1,242 @@
+# Objective
+
+- Provide cached result to Chat Container API calls
+- Compatible to old API interface
+
+# Overview
+
+Container contains a local store for caching resources. The local store caches resources by resource id.
+
+### Store Update
+
+When SDK recieves result from server, for each record, the newly fetched record would always replace the old one with the same record id if there is one.
+
+### API
+
+For fetch API of each resource, there would be a new callback that should return result from cache immediately.
+
+The interface for each language may be different.
+
+```
+getResource(params, callback, cachedCallback?)
+```
+
+Resource includes conversation and message.
+
+The cachedCallback should be optional, thus compatible to the old API interace.
+
+### Expected UI Display / Common Use Case
+
+Client App is expected to keep a single list of messages for display in table. And they are required to keep track to the newest and oldest fetched messages to determine when to fetch result from server.
```

Comment:
Should not need to? I can image with the current API, to keep the message list up-to-date, client apps should only need to:

1. call subscribe() and update their message list in real time.
2. call getMessages() everytime when pubsub are connected. (Do we have need a API callback for that?)